test/tests/functional/pbs_equiv_classes.py

```diff
@@ -1698,20 +1698,24 @@ def test_multiple_job_preemption_order(self):
 
         self.server.deljob([jid1, jid2, jid5])
 
-        # deljob(wait=True) starts the scheduling cycle if job
-        # takes longer to be deleted.
-        # The while loop below is to check that the jobs have been deleted.
-        deleted = 0
+        # if we use deljob(wait=True) starts the scheduling cycle if job
+        # takes more time to be deleted.
+        # The while loop below is to check that the jobs have been deleted
+        # without kicking off a new scheduling cycle.
+
+        deleted = False
         for _ in range(20):
-            if self.server.expect(QUEUE, id=jid1, op=UNSET, max_attempts=None)\
-                    and self.server.expect(QUEUE, id=jid2, op=UNSET,
-                                           max_attempts=None) and\
-                    self.server.expect(QUEUE, id=jid5,
-                                       op=UNSET, max_attempts=None):
-                deleted = 1
+            workq_dict = self.server.status(QUEUE, id='workq')[0]
+            expressq_dict = self.server.status(QUEUE, id='expressq')[0]
+
+            if workq_dict.get('total_jobs') == '2'\
+                    and expressq_dict.get('total_jobs') == '1':
```

Comment:
FYI, I usually see people directly index into a dictionary rather than using the get() method.  So this would be workq_dict['total_jobs'].  They are pretty much the same though.  For consistencies sake, I'd like to see you directly index, but it is your choice.